src/content/en/2021/performance.md

```diff
@@ -1,20 +1,534 @@
+
+{{ figure_markup(
+  image="performance-TTFB-by-rank.png",
+  caption="TTFB performance by rank",
+  description="Stacked bar chart showing TTFB performance for each ranking group. For the top 1,000 websites, 32% had good, 63% needs improvement, and 5% poor performance. For the top 10,000, 27% were good, 66% needs improvement, and 7% poor. For the top 100,000, 22% were good, 65% needs improvement, and 13% poor. For the top 1,000,000, 18% were good, 62% needs improvement, and 20% poor. For all, 18% were good, 58% needs improvement, and 24% poor.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQj6-t49BNV_5W-w83AABkUoo6klfyUyIz13yKShLAzK8qGs5lJ9TKcggIEp6JgxikVF-UJBAHpsrNl/pubchart?oid=829561432&format=interactive",
+  sheets_gid="730149797",
+  sql_file="web_vitals_by_rank.sql"
+  )
+}}
+
+For rank, TTFB was faster for higher-ranking sites. One reason could be that most of these are larger companies with more resources to prioritize performance. They may focus on improving server-side performance and delivering assets through edge CDNs. Another reason could be selection bias - the top origins might be accessed more in regions with closer servers, i.e., lower latency.
+
+<!-- TODO: validate CMS chapter link works -->
+One more possibility has to do with CMS adoption. The [CMS Chapter](./cms) shows CMS adoption by rank.
+
+{{ figure_markup(
+  image="cms-adoption-by-rank.png",
+  caption="CMS adoption by rank",
+  description="Bar chart showing percent of websites using a CMS for each ranking category. On mobile, the numbers are 7% of websites in the top 1,000, 15% in the top 10,000, 20% in the top 100,000, 29% in the top 1,000,000, and 42% in the top 10,000,000.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRzYkQYaBTpD5FKloDXsdb32Y2pjdxFZq_LoekECgTy53F-dRATm9wFFJ3dDVDS8_cGn0h2mKDBjgdM/pubchart?oid=409766380&format=interactive",
+  sheets_gid="730149797",
+  sql_file="../cms/cms_adoption_by_rank.sql"
+  )
+}}
+
+42% of pages (mobile) in the "all" group used a CMS whereas the top 1,000 sites only had 7% adoption.
+
+Then, if we look at the top 5 CMSs by rank, we see that WordPress has the highest adoption at for 33.6% of "all" pages:
+
+{{ figure_markup(
+  image="top-cmss-by-rank.png",
+  caption="Top 5 CMSs by rank",
+  description="Bar chart showing the top CMSs percent of share of each ranking category. WordPress had 3.1% of the top 1,000, 8.6% of the top 10,000, 13.2% of the top 100,000, 21.1% of the top 1,000,000, and 33.6% of the top 10,000,000. The remaining CMSs had less than 3% share of all rankings.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRzYkQYaBTpD5FKloDXsdb32Y2pjdxFZq_LoekECgTy53F-dRATm9wFFJ3dDVDS8_cGn0h2mKDBjgdM/pubchart?oid=2087254046&format=interactive",
+  sheets_gid="730149797",
```

Comment:
Needs to be the full URL as not in this chapter's sheet:

```suggestion
  sheets_gid="https://docs.google.com/spreadsheets/d/1gAJh4VcSEU6-6aQxBiTFl-7cbyHukfdhg-Iaq9y5pnc/#gid=670045665",
```

If there's any other borrowed sheets then similar will need to be done.